packages/graphhopper/index.ts

```diff
@@ -24,6 +24,14 @@ import {
     GraphHopperRouteResponse,
 } from "./parameters"
 
+
+// GraphHopper API returns a `message` property in the response body when an error occurs. 
+interface GraphHopperErrorProps{
+    status_code: number
+    status: string
+    message: string
```

Comment:
When error handling, we should distinguish between the information returned by the server and information returned from `routingjs` to the user. The information returned from GraphHopper looks like this: 
```ts
type GraphHopperHint = {
    message: string
    // not in the docs, but found it when testing locally, so treating these as optional
    details?: string
    point_index?: number
}

type GraphHopperErrorResponseProps = {
    message: string;
    hints: GraphHopperHint[]
}
``` 

We can use that here: 


```ts
  //.request(...)
  //.then(...)
  .catch((error: AxiosError<GraphHopperErrorResponseProps>) => {
      // decide what to do with the response here
  })
```
So we should probably include these hints somewhere in our error properties when we throw. 

Also, let's extend this interface from our base `ErrorProps` so we don't have to repeat the properties common across all browsers.